server/pulp/server/db/model/__init__.py

```diff
@@ -762,6 +763,43 @@ def import_content(self, path, location=None):
         with FileStorage() as storage:
             storage.put(self, path, location)
 
+    def save_and_import_content(self, path, location=None):
+        """
+        Saves this object to the database, then calls safe_import_content.
+
+        :param path: The absolute path to the file to be imported
+        :type path: str
+        :param location: The (optional) location within the unit storage path
+            where the content is to be stored.
+        :type location: str
+        """
+        self.save()
+        self.safe_import_content(path, location)
+
+    def safe_import_content(self, path, location=None):
+        """
+        If import_content raises exception, cleanup then
```

Comment:
s/then continue raising/and raise/
This could probably be on 1 line, Pulp has a line width limit of 100